lang-java/src/main/java/org/eclipse/steady/java/SpringBootAnalyzer.java

```diff
@@ -0,0 +1,635 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.jar.JarEntry;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.steady.ConstructId;
+import org.eclipse.steady.FileAnalysisException;
+import org.eclipse.steady.FileAnalyzer;
+import org.eclipse.steady.core.util.CoreConfiguration;
+import org.eclipse.steady.java.monitor.ClassVisitor;
+import org.eclipse.steady.shared.connectivity.Service;
+import org.eclipse.steady.shared.util.FileSearch;
+import org.eclipse.steady.shared.util.FileUtil;
+import org.eclipse.steady.shared.util.VulasConfiguration;
+
+import javassist.CannotCompileException;
+import javassist.CtClass;
+import javassist.NotFoundException;
+
+/**
+ * Analyzes a single Spring Boot application (JAR) as to identify (and potentially instrument) all its classes (in directory BOOT-INF/classes),
+ * as well as its JARs (in directory WEB-INF/lib).
+ */
+@NotThreadSafe
+public class SpringBootAnalyzer extends JarAnalyzer {
+
+  private static final Logger log =
+      org.apache.logging.log4j.LogManager.getLogger(SpringBootAnalyzer.class);
+
+  private static final String INCL_SPACE = "vulas.core.instr.static.inclSpace";
+  private static final String INCL_BACKEND_URL = "vulas.core.instr.static.inclBackendUrl";
+
+  //	private static final ClassPool CLASSPOOL = ClassPool.getDefault();
+  //	/**
+  //	 * Adds a given URL to the classpath of the class pool. This allows maintaining dependencies
+  // needed for the compilation of instrumented classes.
+  //	 * @param _url
+  //	 * @throws NotFoundException
+  //	 */
+  //	public static void insertClasspath(String _url) throws NotFoundException {
+  // CLASSPOOL.insertClassPath(_url); }
+  //	private static ClassPool getClassPool() { return WarAnalyzer.CLASSPOOL; }
+
+  //	private static boolean OVERWRITE_ENABLED = false;
+  //	public static void enableOverwrite(boolean _b) { WarAnalyzer.OVERWRITE_ENABLED = _b; }
+  //	public static boolean isOverwriteEnabled() { return WarAnalyzer.OVERWRITE_ENABLED; }
+
+  private Map<JavaId, ClassVisitor> instrumentedClasses = new HashMap<JavaId, ClassVisitor>();
+  private Path tmpDir = null; // To where the JAR is extracted
+  private Path inclDir = null;
+
+  private ArchiveAnalysisManager mgr = null;
+  private Set<FileAnalyzer> nestedAnalyzers = null;
+
+  /** The Vulas archive in incl/ will be rewritten as to contain a proper vulas configuration file. The original will
+   * then need to be skipped, i.e., not included in the instrumented WAR. */
```

Comment:
i think this comment is a left-over of the WarAnalyzer

Reply:
Yes, but the same principle is applied here.